[quant][graphmode][fx] Add support for specifying reference quantized module mapping in backend_config_dict

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #68770
* #68769
* #68229
* #68228
* __->__ #68227

Summary:
This PR adds two keys to backend_config_dict:
"root_module": the root module for the pattern (since we may have patterns for fused ops)
"reference_quantized_module_for_root": the corresponding reference quantized module for the root

Test Plan:
```
python test/test_quant_trt.py TestQuantizeFxTRTOps
python test/test_quant_trt.py TestConvertFxDoNotUse
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D32537711](https://our.internmc.facebook.com/intern/diff/D32537711)

jerryzh168 committed:
[quant][graphmode][fx] Add support for specifying reference quantized module mapping in backend_config_dict

Summary:
This PR adds two keys to backend_config_dict:
"root_module": the root module for the pattern (since we may have patterns for fused ops)
"reference_quantized_module_for_root": the corresponding reference quantized module for the root

Test Plan:
```
python test/test_quant_trt.py TestQuantizeFxTRTOps
python test/test_quant_trt.py TestConvertFxDoNotUse
```

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx] Add support for specifying reference quantized module mapping in backend_config_dict"


Summary:
This PR adds two keys to backend_config_dict:
"root_module": the root module for the pattern (since we may have patterns for fused ops)
"reference_quantized_module_for_root": the corresponding reference quantized module for the root

Test Plan:
```
python test/test_quant_trt.py TestQuantizeFxTRTOps
python test/test_quant_trt.py TestConvertFxDoNotUse
```

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx] Add support for specifying reference quantized module mapping in backend_config_dict"


Summary:
This PR adds two keys to backend_config_dict:
"root_module": the root module for the pattern (since we may have patterns for fused ops)
"reference_quantized_module_for_root": the corresponding reference quantized module for the root

Test Plan:
```
python test/test_quant_trt.py TestQuantizeFxTRTOps
python test/test_quant_trt.py TestConvertFxDoNotUse
```

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented:
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32537711).

jerryzh168 committed:
Update on "[quant][graphmode][fx] Add support for specifying reference quantized module mapping in backend_config_dict"


Summary:
This PR adds two keys to backend_config_dict:
"root_module": the root module for the pattern (since we may have patterns for fused ops)
"reference_quantized_module_for_root": the corresponding reference quantized module for the root

Test Plan:
```
python test/test_quant_trt.py TestQuantizeFxTRTOps
python test/test_quant_trt.py TestConvertFxDoNotUse
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D32537711](https://our.internmc.facebook.com/intern/diff/D32537711)

[ghstack-poisoned]

vkuzo commented on torch/ao/quantization/fx/backend_config_dict/tensorrt.py:
```diff
@@ -31,7 +31,9 @@ def get_tensorrt_backend_config_dict():
         "dtype_configs": [
             weighted_op_qint8_dtype_config,
-        ]
+        ],
```
should we be adding some guidance for users on what this means along the way?

jerryzh168 replied:
Sure, I can add some comments for now? will have a more complete README later when these things are more stable

jerryzh168 committed:
Update on "[quant][graphmode][fx] Add support for specifying reference quantized module mapping in backend_config_dict"


Summary:
This PR adds two keys to backend_config_dict:
"root_module": the root module for the pattern (since we may have patterns for fused ops)
"reference_quantized_module_for_root": the corresponding reference quantized module for the root

Test Plan:
```
python test/test_quant_trt.py TestQuantizeFxTRTOps
python test/test_quant_trt.py TestConvertFxDoNotUse
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D32537711](https://our.internmc.facebook.com/intern/diff/D32537711)

[ghstack-poisoned]

jerryzh168 commented:
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32537711).

jerryzh168 committed:
Update on "[quant][graphmode][fx] Add support for specifying reference quantized module mapping in backend_config_dict"


Summary:
This PR adds two keys to backend_config_dict:
"root_module": the root module for the pattern (since we may have patterns for fused ops)
"reference_quantized_module_for_root": the corresponding reference quantized module for the root

Test Plan:
```
python test/test_quant_trt.py TestQuantizeFxTRTOps
python test/test_quant_trt.py TestConvertFxDoNotUse
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D32537711](https://our.internmc.facebook.com/intern/diff/D32537711)

[ghstack-poisoned]

jerryzh168 commented:
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32537711).

jerryzh168 committed:
Update on "[quant][graphmode][fx] Add support for specifying reference quantized module mapping in backend_config_dict"


Summary:
This PR adds two keys to backend_config_dict:
"root_module": the root module for the pattern (since we may have patterns for fused ops)
"reference_quantized_module_for_root": the corresponding reference quantized module for the root

Test Plan:
```
python test/test_quant_trt.py TestQuantizeFxTRTOps
python test/test_quant_trt.py TestConvertFxDoNotUse
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D32537711](https://our.internmc.facebook.com/intern/diff/D32537711)

[ghstack-poisoned]

jerryzh168 commented:
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32537711).
